### PR TITLE
Auth ui

### PR DESCRIPTION
**Pull request by eliasson**
Initial work on authentication. A hybrid solution where registration, activation, etc is implemented using django views and templates whereas the login and actual QuarterApp application is implemented using Angular.

Not yet complete, since angular routes checking public / private is fully implemented. Still think this can go as it enables quite a few pieces in the toolchain (including activating REST API and CORS in ACME reference project).

### CODE REVIEW

File: quarterapp/accounts/forms.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+
+class ForgotPasswordForm(forms.Form):
+    """
+    Form used to create an object with a random code used to reset the users
+    password. This form does not reset the password, it only creates the code
+    and sends an email to the user.
+    """
+    username = forms.CharField(label=_('Username'))
+    # TODO Should we allow both username or email address for reset?
```
Comment (dbackman) [inline quarterapp/accounts/forms.py]: We have an email auth backend, one can login with email. To be consistent we should allow email for reset as well. (Fix it later)

File: quarterapp/accounts/tests/test_password.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from django.core.urlresolvers import reverse
+from django.contrib.auth import authenticate
+from django.test import TestCase
+
+from quarterapp.accounts.tests import factories
```
Comment (dbackman) [inline quarterapp/accounts/tests/test_password.py]: Relative imports please

File: quarterapp/ui/static/app/auth/login.controller.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+        function submit() {
+            if (vm.loginForm.$valid) {
+                auth.login(vm.loginData.username, vm.loginData.password)
+                    .then(function () {
+                        $location.url('/');
+                    },
+
+                    function () {
+                        vm.invalidCredenials = true;
```
Comment (dbackman) [inline quarterapp/ui/static/app/auth/login.controller.js]: Credenials misspelled, should be "Creden**t**ials"

File: quarterapp/ui/templates/ui/accounts/register.html [hunk 4]
```diff
@@ -0,0 +1,73 @@
+{% extends "ui/pages/_base.html" %}
+{% load i18n %}
+
+{% block page_title %}
+    {% trans 'Register new account' context 'Page title' %} - QuarterApp
+{% endblock page_title %}
+
+
+{% block page %}
```
Comment (dbackman) [inline quarterapp/ui/templates/ui/accounts/register.html]: General comment, why not use form.field.label_tag instead writing the <label for=>...</label> ? (https://docs.djangoproject.com/en/1.9/ref/forms/api/#django.forms.BoundField.label_tag)

File: quarterapp/ui/views.py [hunk 5]
```diff
@@ -18,7 +18,15 @@
 
 from __future__ import unicode_literals
```
Comment (dbackman) [inline quarterapp/ui/views.py]: Needed?

File: quarterapp/ui/views.py [hunk 6]
```diff
@@ -18,7 +18,15 @@
...
 
-from django.views.generic import TemplateView
+from django.contrib.auth.forms import AuthenticationForm
+from django.core.urlresolvers import reverse, reverse_lazy
+from django.shortcuts import redirect
+from django.views.generic import TemplateView, FormView, View
+from django.contrib.auth import logout, login, authenticate
+
+from quarterapp.accounts.forms import ForgotPasswordForm, ResetPasswordForm, \
```
Comment (dbackman) [inline quarterapp/ui/views.py]: Relative imports please

File: quarterapp/ui/views.py [hunk 7]
```diff
@@ -29,9 +37,91 @@ class AppView(TemplateView):
...
+        return super(ResetPasswordView, self).form_valid(form)
+
+
+class ConfirmEmailChangeView(TemplateView):
+    """
+    Issues the email change if the given confirmation code is a correct code.
+    """
+    template_name = 'ui/accounts/email_changed.html'
+
+    def get(self, *args, **kwargs):
```
Comment (dbackman) [inline quarterapp/ui/views.py]: It violates the REST principles for GET not modifying states...

**Comment by dbackman** [comment]
Good work! Consider my comments. Maybe create issues for some of it or at least create TODO:s? Then ok to merge.

File: quarterapp/accounts/tests/test_password.py (hunk 2 above)
Comment (eliasson) [inline quarterapp/accounts/tests/test_password.py]: Fixed in 25249c091c6e520d5ac1f54e93df06a5fe3c5572

File: quarterapp/ui/views.py (hunk 5 above)
Comment (eliasson) [inline quarterapp/ui/views.py]: Removed in 25249c091c6e520d5ac1f54e93df06a5fe3c5572

File: quarterapp/ui/views.py (hunk 6 above)
Comment (eliasson) [inline quarterapp/ui/views.py]: Fixed in 25249c091c6e520d5ac1f54e93df06a5fe3c5572

File: quarterapp/ui/static/app/auth/login.controller.js (hunk 3 above)
Comment (eliasson) [inline quarterapp/ui/static/app/auth/login.controller.js]: Fixed in 235dcbc220ee6972fd40d44db8aee4d2557d3703

File: quarterapp/ui/templates/ui/accounts/register.html (hunk 4 above)
Comment (eliasson) [inline quarterapp/ui/templates/ui/accounts/register.html]: No good reason, use label_tag in ee98d07cd2bc514918dc4808920c470485a11b56

File: quarterapp/ui/views.py (hunk 7 above)
Comment (eliasson) [inline quarterapp/ui/views.py]: Agree, created #8 to cover this need.

**Pull request merged**